File: fixtures/sample.ts
// Penumbra fixture — TypeScript (tint: blue)
import { useState, useEffect } from "react";

interface User {
  id: number;
  name: string;
  active: boolean;
  tags: string[];
}

type Result<T> = { ok: true; value: T } | { ok: false; error: string };

export const greet = async (user: User): Promise<Result<string>> => {
  if (!user.active) return { ok: false, error: "inactive" };
  await new Promise((r) => setTimeout(r, 100));
  return { ok: true, value: `Hello, ${user.name}` };
};

const users: readonly User[] = [
  { id: 1, name: "Ada",   active: true,  tags: ["admin"] },
  { id: 2, name: "Grace", active: false, tags: [] },
];

// Test: regex, numbers, escape sequences
const re = /^hello\s+(\w+)$/gi;
const n = 0xff_ff + 1e3;
console.log(`matched ${re.test("hello world")}, n=${n}`);

File: fixtures/sample.rs
// Penumbra fixture — Rust (tint: orange)
use std::collections::HashMap;

#[derive(Debug, Clone)]
pub struct User {
    pub id: u32,
    pub name: String,
    pub active: bool,
}

#[derive(Debug)]
pub enum GreetError {
    Inactive,
    Empty,
}

impl User {
    pub fn greet(&self) -> Result<String, GreetError> {
        if !self.active {
            return Err(GreetError::Inactive);
        }
        if self.name.is_empty() {
            return Err(GreetError::Empty);
        }
        Ok(format!("Hello, {}", self.name))
    }
}

fn main() {
    let users: Vec<User> = vec![
        User { id: 1, name: "Ada".into(),   active: true  },
        User { id: 2, name: "Grace".into(), active: false },
    ];

    let mut tally: HashMap<bool, u32> = HashMap::new();
    for u in &users {
        *tally.entry(u.active).or_insert(0) += 1;
    }
    println!("{:#?}", tally);
}
